commands/commands.go

```diff
@@ -211,13 +214,18 @@ func (c *Command) waitForTimeout() error {
 		defer func() { quit <- 0 }()
 	}
 	log.Debugf("%s.run waiting for PID %d: ", c.Name, cmd.Process.Pid)
-	waitStatus, err := cmd.Process.Wait()
-	if waitStatus != nil && !waitStatus.Success() {
-		return fmt.Errorf("%s exited with error", c.Name)
-	} else if err != nil {
-		if err.Error() == errNoChild {
-			log.Debugf(err.Error())
-			return nil // process exited cleanly before we hit wait4
+
+	defer reapChildren(c.Cmd.SysProcAttr.Pgid)
+	err := c.Cmd.Wait()
+	if err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+				if status.ExitStatus() == 0 {
+					log.Debug(err)
```

Comment:
Oops, no. Looks like there a couple places I left a bit of debugging. Will clean that up.